agent.py

```diff
@@ -32,7 +36,10 @@ def main(
             f"Found {len(available_markets)} markets: {[m.question for m in available_markets]}"
         )
 
-        market = agent.pick_market(available_markets)
+        market = t.cast(
+            t.Union[manifold.ManifoldMarket, omen.OmenMarket],
+            agent.pick_market(available_markets),
+        )  # TODO: Mypy bug: Works in VSCode type-checking, but doesn't in Mypy.
```

Comment:
Does
```
market: t.Union[manifold.ManifoldMarket, omen.OmenMarket] = agent.pick_market(available_markets)
```
not work?

Reply:
It doesn't :( I'm not sure what's Mypy's problem here, with Pylance (Pyright) it's just okay: 

![Screenshot by Dropbox Capture](https://github.com/evangriffiths/prediction-market-agent/assets/15619339/14ae143a-6b4f-424b-9486-b0b616c2a3db)

Reply:
Not needed anymore, thanks to the removal of Protocol! 🎉